pkg/controller/operators/catalog/operator.go

```diff
@@ -142,6 +148,32 @@ func NewOperator(ctx context.Context, kubeconfigPath string, clock utilclock.Clo
 		return nil, err
 	}
 
+	var restConfig *rest.Config
```

Comment:
nit: Can we move the below logic into a standalone function?

If you see above, we have things like the `opClient` and `dynamicClient` that are returned from `NewForConfig` functions. Using that same pattern here would help the readability IMO